scripts/fastlane/Fastfile

```diff
@@ -0,0 +1,157 @@
+RELEASE_TAG_PREFIX = 'release_v'
+default_platform(:android)
+
+platform :android do |options|
+  gradle_properties_path = File.expand_path("#{Dir.pwd()}/../../gradle.properties")
+  gradle_project_root = File.dirname(gradle_properties_path)
+  change_log_path = "#{gradle_project_root}/CHANGELOG.md"
+  version_regex = /(\d*)\.(\d*)\.(\d*)(?:-unstable\.)?(\d*)?/
+  pr_body = %Q(
+## Release PR review checklist
+- [ ] Verify version name in gradle.properties
+- [ ] Verify CHANGELOG.md
+)
+
+  desc "Fetch all tags to ensure accurate calculation of the next release number."
+  before_all do 
+    sh('git', 'fetch', '--tags')
+    sh('git', 'fetch')
+  end
+
+  lane :configure_git_options do |options|
+    sh('git', 'config', 'user.email', options[:git_user_email])
+    sh('git', 'config', 'user.name', options[:git_user_name])
+  end
+
+  lane :manage_next_release_pr do |options|
+    product_name = options[:product_name]
+    last_release_tag = last_git_tag(pattern: "#{RELEASE_TAG_PREFIX}*")
+    next_version, commits = calculate_next_release_version(release_tag_prefix: RELEASE_TAG_PREFIX, from_tag: last_release_tag)
+    UI.message("Version change #{last_release_tag} => #{next_version}")
+    update_gradle_properties(version: next_version)
+    update_change_log(last_release_tag: last_release_tag)
+
+    git_add(path: [gradle_properties_path, change_log_path])
+    git_commit(path: [gradle_properties_path, change_log_path], message: "chore(release): #{product_name} - #{next_version}")
+    push_to_git_remote(force: true)
+
+
+    current_branch = sh('git', 'branch', '--show-current')
+    repo_name_parts = options[:repo].split('/')
+    create_pull_request(base: 'main', 
+                        title: "chore(release): #{product_name} - #{next_version}", 
+                        body:pr_body, 
+                        repo: options[:repo], 
+                        api_token: ENV["RELEASE_MANAGER_TOKEN"])
+  end
+
+
+  desc "Checks out a tag and runs the uploadArchives gradle task."
+  lane :push_tag_to_maven do |options|
+    UI.message("Pushing release #{options[:release_tag]} to maven")
+    sh("git", "checkout", options[:release_tag])
+    publish_to_maven
+  end
```

Comment:
I don't think we should create the tag until after the `uploadArchives` command has succeeded.   That way, if `uploadArchives` fails for some reason, we don't have to worry about rolling back (deleting the tag).   And even then, we should just figure out how we can make a GitHub release, which should automatically create the tag as a part of that. 

```suggestion
```

Reply:
That's correct. Based on our discussion, we will use the github API to create a release. I just haven't removed the code.